recipes/sagemaker-code-editor/meta.yaml

```diff
@@ -0,0 +1,60 @@
+{% set name = "sagemaker-code-editor" %}
+{% set version = "0.1.0" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version|replace("-", "_") }}
+
+source:
+  url: https://sagemaker-code-editor-1696292125.s3.us-west-2.amazonaws.com/code-editor.tar.gz
+  sha256: 9effda827c45e1fd8dfc534ce310791ee46070d818ede89abf19e3ec9c8906d0
+  folder: sagemaker-code-editor
+
+build:
+  number: 0
+  skip: true  # [not linux]
+
+requirements:
+  build:
+    - sysroot_linux-64 ==2.17
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - make
+    - pkg-config
+    - nodejs >=18.15,<19
+    - openssl ==3.1.4
```

Comment:
What's the reason for this strict pin? It will eventually break, so you might as well add a comment on why it's needed, or remove if it's not.

Reply:
Actually, a lot of these dependencies should (very likely) not be in the build environment, but in host.

Reply:
I'll check if the openssl pin is needed

>Actually, a lot of these dependencies should (very likely) not be in the build environment, but in host.

@h-vetinari - I want to understand this better. While I look up the [docs](https://docs.conda.io/projects/conda-build/en/stable/resources/define-metadata.html#requirements-section) - it'll help if you can also please provide a bit more detail here or example of a dependency that should be in host instead? Thank you.

Reply:
> I want to understand this better. While I look up the [docs](https://docs.conda.io/projects/conda-build/en/stable/resources/define-metadata.html#requirements-section) - it'll help if you can also please provide a bit more detail here or example of a dependency that should be in host instead? Thank you.

the distinction between build/host/run is one of the things that people regularly struggle with the longest. I recently wrote another [comment](https://github.com/conda-forge/staged-recipes/pull/24841#discussion_r1452946898) trying to explain it, hope it helps.

Reply:
thank you, going through this

Reply:
@h-vetinari , @jaimergp  I removed the openssl dependency altogether - also looking at other dependencies to confirm they're in the right section

Reply:
just noting - krb5 is needed in build dependencies also

Removing it from build had below error:  
```
make: Entering directory '/home/conda/staged-recipes/build_artifacts/sagemaker-code-editor_1707759366510/work/sagemaker-code-editor/src/remote/node_modules/kerberos/build'
  CXX(target) Release/obj.target/kerberos/src/kerberos.o
  CXX(target) Release/obj.target/kerberos/src/unix/base64.o
  CXX(target) Release/obj.target/kerberos/src/unix/kerberos_gss.o
  CXX(target) Release/obj.target/kerberos/src/unix/kerberos_unix.o
  SOLINK_MODULE(target) Release/obj.target/kerberos.node
/home/conda/staged-recipes/build_artifacts/sagemaker-code-editor_1707759366510/_build_env/bin/../lib/gcc/x86_64-conda-linux-gnu/12.3.0/../../../../x86_64-conda-linux-gnu/bin/ld: cannot find -lkrb5: No such file or directory
/home/conda/staged-recipes/build_artifacts/sagemaker-code-editor_1707759366510/_build_env/bin/../lib/gcc/x86_64-conda-linux-gnu/12.3.0/../../../../x86_64-conda-linux-gnu/bin/ld: cannot find -lgssapi_krb5: No such file or directory
collect2: error: ld returned 1 exit status
make: *** [kerberos.target.mk:144: Release/obj.target/kerberos.node] Error 1
make: Leaving directory '/home/conda/staged-recipes/build_artifacts/sagemaker-code-editor_1707759366510/work/sagemaker-code-editor/src/remote/node_modules/kerberos/build'

```

Reply:
I've moved `libsecret ` to the host section and I've added comments to categorize all dependencies.
Please let me know if there are any more suggestions.